From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: gary_gen <gary_gen@realtek.com>
Date: Wed, 22 Jul 2026 00:00:00 +0800
Subject: [PATCH] arm64: dts: enable hdcp2 on bleedingedge-rtd1619b

Enable HDCP2.x on bleedingedge-rtd1619b board.

Set hdcp = <2> for HDCP2.x support and ext-flags = <0x40000>
to defer HDCP2 TEE init until userspace triggers HDCP.

Remove ext-flags = <0x0> override from board-level DTS files so
they inherit ext-flags from rtd1619b-bleedingedge.dtsi.

Signed-off-by: gary_gen <gary_gen@realtek.com>
---
 arch/arm64/boot/dts/realtek/rtd1619b-backinblack-2gb.dts  | 4 ----
 arch/arm64/boot/dts/realtek/rtd1619b-bleedingedge-2gb.dts | 4 ----
 arch/arm64/boot/dts/realtek/rtd1619b-bleedingedge-4gb.dts | 4 ----
 arch/arm64/boot/dts/realtek/rtd1619b-bleedingedge.dtsi    | 3 ++-
 4 files changed, 2 insertions(+), 13 deletions(-)

diff --git a/arch/arm64/boot/dts/realtek/rtd1619b-backinblack-2gb.dts b/arch/arm64/boot/dts/realtek/rtd1619b-backinblack-2gb.dts
index 5ab74c589ff2..9bb306f33f8f 100644
--- a/arch/arm64/boot/dts/realtek/rtd1619b-backinblack-2gb.dts
+++ b/arch/arm64/boot/dts/realtek/rtd1619b-backinblack-2gb.dts
@@ -93,10 +93,6 @@ panel_in_dsi: endpoint {
 	};
 };

-&hdmi {
-	ext-flags = <0x0>;
-};
-
 &sound {
 	compatible = "realtek,rtk-alsa-hifi";
 };
diff --git a/arch/arm64/boot/dts/realtek/rtd1619b-bleedingedge-2gb.dts b/arch/arm64/boot/dts/realtek/rtd1619b-bleedingedge-2gb.dts
index 731f965d081b..bf4231fe95d6 100644
--- a/arch/arm64/boot/dts/realtek/rtd1619b-bleedingedge-2gb.dts
+++ b/arch/arm64/boot/dts/realtek/rtd1619b-bleedingedge-2gb.dts
@@ -44,10 +44,6 @@ &voext {
 	status = "disabled";
 };

-&hdmi {
-	ext-flags = <0x0>;
-};
-
 &sound {
 	compatible = "realtek,rtk-alsa-hifi";
 };
diff --git a/arch/arm64/boot/dts/realtek/rtd1619b-bleedingedge-4gb.dts b/arch/arm64/boot/dts/realtek/rtd1619b-bleedingedge-4gb.dts
index fd370e35bc09..533ad2ac177c 100644
--- a/arch/arm64/boot/dts/realtek/rtd1619b-bleedingedge-4gb.dts
+++ b/arch/arm64/boot/dts/realtek/rtd1619b-bleedingedge-4gb.dts
@@ -49,10 +49,6 @@ &voext {
 	status = "disabled";
 };

-&hdmi {
-	ext-flags = <0x0>;
-};
-
 &sound {
 	compatible = "realtek,rtk-alsa-hifi";
 };
diff --git a/arch/arm64/boot/dts/realtek/rtd1619b-bleedingedge.dtsi b/arch/arm64/boot/dts/realtek/rtd1619b-bleedingedge.dtsi
index b6d769963548..401c3c500b62 100644
--- a/arch/arm64/boot/dts/realtek/rtd1619b-bleedingedge.dtsi
+++ b/arch/arm64/boot/dts/realtek/rtd1619b-bleedingedge.dtsi
@@ -263,7 +263,8 @@ &display_subsystem {

 &hdmi {
 	hdmi5v-gpios = <&gpio 5 GPIO_ACTIVE_LOW>;
-	hdcp = <0>;
+	hdcp = <2>;
+	ext-flags = <0x40000>;
 };

 &cvbs {
--
2.34.1
